### DISCUSSION TITLE
Bulk Actions and Error Handling -- What are our options?

### DISCUSSION THREAD

**Discussion by SimonMacIntyre** [rappasoft/laravel-livewire-tables#1035] [Q&A] [answered]
Hi there, I have made a bulk action. 

Naturally, an action has a chance to encounter an error in the backend. However, I notice throwing an exception has no effect at all on the table or view. Secondly, I could not find anything in the documentation about this. 

Preferably I would be able to put a single error message rendering at the top of the table, a dismissible alert. But the Custom View can only be appended at the bottom from what I can see, and although there might be a solution involving `modals` to display errors, I really would rather a simple text in a dismissible form.

Just wondering, is Custom View the only option here in terms of error handling? Basically populate some `$error` and `Custom View` can render it? I just wish it could be above the table instead of below so it is not limited to basically only modals and nothing else. 

Another option I am thinking of is maybe flashing a session message about it and rendering it completely outside of the livewire table entirely. 

Just hoping to hear some other ideas or point me in the right direction if anyone has any advice, thanks!

**Comment by lrljoe** [comment, marked as answer]
You can add a view to any of the following configurable areas.  before-toolbar is likely to be your friend here!

```
public function configure(): void
{
  $this->setConfigurableAreas([
    'toolbar-left-start' => 'path.to.my.view',
    'toolbar-left-end' => 'path.to.my.view',
    'toolbar-right-start' => 'path.to.my.view',
    'toolbar-right-end' => 'path.to.my.view',
    'before-toolbar' => 'path.to.my.view',
    'after-toolbar' => 'path.to.my.view',
    'before-pagination' => 'path.to.my.view',
    'after-pagination' => 'path.to.my.view',
  ]);
}
```
There's a map of where they relate to at the bottom of this page:

[https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/configurable-areas](https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/configurable-areas)

**Reply by SimonMacIntyre** [reply]
Hmmm interesting, I see that now. Why is `Custom View` segregated in the docs the way it is 🤔  It does seem the same functionally as the PR I made.  Now that I see this, it seems to me like `Custom View` is just better suited to be moved inside this under a key like `after-table`. So the question still on my mind is, would a `before-table` be any different than `before-toolbar`?

**Comment by SimonMacIntyre** [comment]
https://github.com/rappasoft/laravel-livewire-tables/pull/1036 

I made this PR and tested it and am using it now in my project.

**Comment by lrljoe** [comment]
So often the way!  I've got a couple of other bits that I'll be doing PRs for, so I'll add a link into the documentation to/from the customView and Configurable Areas.

I imagine a before-table would come above the <tools> section, whereas before toolbar comes after the sorting/filter pils, but before the toolbar itself.

I think Anthony will be more likely to accept a PR for that than for your current PR, as another customView will probably add to the confusion!
